triehash/Cargo.toml

```
@@ -1,17 +1,17 @@
 [package]
 name = "triehash"
-version = "0.5.0"
+version = "0.5.1"
 authors = ["Parity Technologies <admin@parity.io>"]
 description = "In-memory patricia trie operations"
 repository = "https://github.com/paritytech/parity-common"
 license = "GPL-3.0"
 
 [dependencies]
-hash-db = { version = "0.11.0", default-features = false }
+hash-db = { version = "0.12.3", default-features = false, features = ["std"] }
```

This implies `std` for all consuming code I think. Is that ok? The reason seem to be to have `Debug` impl'd.

---

triehash is currently not `no_std` compatible so it seems coherent.
`trie` crates support `no_std` so it could also make sense to make `triehash` no_std compatible.

---

I was about to make trie_hash no_std (it is a pretty direct reuse of what is in trie_root crate), but then I realize that it does not make sense without also making rlp no_std.
So I guess it is a bit to much for this pr.

---

Isn't that technically a breaking change? `hash_db::Hasher` from `0.12` is incompatible with the one from `0.11`, right?
Also this can be simplified to 
```suggestion
hash-db = "0.12.3"
```

---

yes indirectly it can be seen as a breaking change, I will update the version accordingly, thanks.